python/server.py

```diff
@@ -66,256 +107,371 @@ def empty_to_none(field):
+            )
+        )
+
+        # create link token
+        response = client.link_token_create(request)
+        return jsonify(response.to_dict())
+    except plaid.ApiException as e:
+        return json.loads(e.body)
+
 
 # Exchange token flow - exchange a Link public_token for
 # an API access_token
 # https://plaid.com/docs/#exchange-token-flow
+
+
 @app.route('/api/set_access_token', methods=['POST'])
 def get_access_token():
-  global access_token
-  global item_id
-  public_token = request.form['public_token']
-  try:
-    exchange_response = client.Item.public_token.exchange(public_token)
-  except plaid.errors.PlaidError as e:
-    return jsonify(format_error(e))
-
-  pretty_print_response(exchange_response)
-  access_token = exchange_response['access_token']
-  item_id = exchange_response['item_id']
-  return jsonify(exchange_response)
+    global access_token
+    global item_id
+    public_token = request.form['public_token']
+    try:
+        exchange_request = ItemPublicTokenExchangeRequest(
+            public_token=public_token)
+        exchange_response = client.item_public_token_exchange(exchange_request)
+        print(exchange_response)
+        access_token = exchange_response['access_token']
+        item_id = exchange_response['item_id']
+        return jsonify(exchange_response.to_dict())
+    except plaid.ApiException as e:
+        return json.loads(e.body)
+
 
 # Retrieve ACH or ETF account numbers for an Item
 # https://plaid.com/docs/#auth
+
+
 @app.route('/api/auth', methods=['GET'])
 def get_auth():
-  try:
-    auth_response = client.Auth.get(access_token)
-  except plaid.errors.PlaidError as e:
-    return jsonify({'error': {'display_message': e.display_message, 'error_code': e.code, 'error_type': e.type } })
-  pretty_print_response(auth_response)
-  return jsonify(auth_response)
+    try:
+        ag_request = AuthGetRequest(
+            access_token=access_token
+        )
+        auth_response = client.auth_get(ag_request)
+        print(auth_response)
+        return jsonify(auth_response.to_dict())
+    except plaid.ApiException as e:
+        error_response = format_error(e)
+        return jsonify(error_response)
+
 
 # Retrieve Transactions for an Item
 # https://plaid.com/docs/#transactions
+
+
 @app.route('/api/transactions', methods=['GET'])
 def get_transactions():
-  # Pull transactions for the last 30 days
-  start_date = '{:%Y-%m-%d}'.format(datetime.datetime.now() + datetime.timedelta(-30))
-  end_date = '{:%Y-%m-%d}'.format(datetime.datetime.now())
-  try:
-    transactions_response = client.Transactions.get(access_token, start_date, end_date)
-  except plaid.errors.PlaidError as e:
-    return jsonify(format_error(e))
-  pretty_print_response(transactions_response)
-  return jsonify(transactions_response)
+    # Pull transactions for the last 2 years
+    START_DATE = (datetime.now() - timedelta(days=(365 * 2)))
+    END_DATE = datetime.now()
```

Comment:
was this updated to 2 years to match other quickstart backends?

Reply:
I dont think these need to be all caps, they are just normal variables here. Usually all caps is some sort of global constant

Reply:
changed back to 30 days, and will put variables in lower case.  this is true in ruby too, so will change there as well.

Reply:
okay, so all the backends now use 30 days for transactions and investment transactions